### DISCUSSION TITLE
Different sources and available fields

### DISCUSSION THREAD

**Discussion by Eran-T** [freelawproject/courtlistener#3280] [Q&A] [answered]
Hi, 
While exploring the API I've noticed that the docket and endpoint each has a "source" field. These fields hold different values for both endpoint and samples relying on some sources tend to consistently have some fields while samples relying on others don't. 

My questions are:
1. Is there any documentation indicating which sources relatively consistently have which fields and which fields are consistently not available from which sources? 

2. since there are several sources is it possible that the same docket/opinion will appear twice from two different sources with no built-in indication that the item is the same in both?

3. specifically for parties involved in a civil case (docket). Is there any indication as to the frequency of missing parties? what are different alternatives to finding the parties involved in a specific case other than searching for the docket number in parties endpoint (if there are any creative alternatives). take for example this docket:
https://www.courtlistener.com/api/rest/v3/dockets/67013430/
when searching for this docket in the parties endpoint, no results come up. 
https://www.courtlistener.com/api/rest/v3/parties/?docket=67013430
maybe the parties involved in this case can be found somewhere else in the API? (here is a link documenting the parties involved in the case).
https://www.docketalarm.com/cases/Delaware_State_Court_of_Chancery/2023-0215/2023%3B_CONF_ORD_-_CONS_W-_2023-0216-MTZ_-_IN_RE_AMC_ENTERTAINMENT_HOLDINGS_INC._STOCKHOLDER_LITIGATION/

thank you very much for the help!

**Comment by mlissner** [comment, marked as answer]
These are good questions. 

> 1. Is there any documentation indicating which sources relatively consistently have which fields and which fields are consistently not available from which sources?

Unfortunately no. We've gathered a lot of data from a lot of places, and we haven't kept documentation of which sources have which data. It would certainly have been a good idea, sorry.

> 2. since there are several sources is it possible that the same docket/opinion will appear twice from two different sources with no built-in indication that the item is the same in both?

Generally no, but we do have duplicates in the system that appear for other reasons:

1. When we import content, we do our best to merge the new content into the old, but the legal space doesn't have good identifiers, so this is imperfect, and mistakes do happen.

2. When we scrape content from court websites, we create duplicates (on purpose), whenever a court updates a PDF for whatever reason. We want to keep every record of every case, so that's why we do this. In general, you'll want the latest version of each scraped opinion.

> 3. specifically for parties involved in a civil case (docket). Is there any indication as to the frequency of missing parties? what are different alternatives to finding the parties involved in a specific case other than searching for the docket number in parties endpoint (if there are any creative alternatives). take for example this docket:
>        https://www.courtlistener.com/api/rest/v3/dockets/67013430/
>        when searching for this docket in the parties endpoint, no results come up.
>        https://www.courtlistener.com/api/rest/v3/parties/?docket=67013430
>        maybe the parties involved in this case can be found somewhere else in the API? (here is a link documenting the parties involved in the case).
>        https://www.docketalarm.com/cases/Delaware_State_Court_of_Chancery/2023-0215/2023%3B_CONF_ORD_-_CONS_W-_2023-0216-MTZ_-_IN_RE_AMC_ENTERTAINMENT_HOLDINGS_INC._STOCKHOLDER_LITIGATION/

That case is in the Delaware Chancery court, which means we only get opinions for it, and not the other types of data, so I think the guiding rule is that if it's a federal case, then there's a good chance we'll have the party information for it. If it's a state case (like this one), then we will not (at least for the time being). 

Some federal cases won't have parties (because everything costs money), so if you run into that, I'd recommend using our Fetch API to get the content into our system.

I hope this helps. Let me know what else comes up!
